woocommerce/woocommerce
# [WC_Tracker] Track woocommerce_allow_tracking_toggled

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR tracks woocommerce_allow_tracking_toggled when woocommerce_allow_tracking is changed in:

- [WooCommerce - Settings - Advanced - woocommerce.com - Allow usage of WooCommerce to be tracked](https://github.com/woocommerce/woocommerce/compare/add/track-woocommerce-allow-tracking?expand=1#diff-b0114fafd767404eb935cc6fb90c59af35071d049638143104733044b9cfe335)
- [Allow WooCommerce to Track Note](https://github.com/woocommerce/woocommerce/compare/add/track-woocommerce-allow-tracking?expand=1#diff-13c5495cafe4403968d77a5a6deeb18e6d87aae998ce4a149acfb79218deb0c2)
- [After WCCOM is connected ](https://github.com/woocommerce/woocommerce/compare/add/track-woocommerce-allow-tracking?expand=1#diff-32b26bf8430b272079f3c981317024f2a7d7848b4198ae7878716cc133d10961R921)
- [When the checkbox is toggled in Onboarding Core Profiler](https://github.com/woocommerce/woocommerce/pull/53715/files#diff-4e3543935fede6f610fd95d6713b23d421d1fcce3a9ee6285f4b98c34b9335b5R130)

PT: pffQ75-1bu-p2



### How to test the changes in this Pull Request:

1. Checkout this PR
2. Enable WC Tracks debug `localStorage.setItem( 'debug', 'wc-admin:*' )`
3. Install and Activate WooCommerce 
4. Go over WooCommerce Onboarding and toggle the checkbox "I agree to share my data to tailor my store setup experience..."
5. See the event `wcadmin_woocommerce_allow_tracking_toggled` being triggered with `core-profiler` context
6. Download and load Tracks Vigilante Chrome extension https://github.com/Automattic/tracks-chrome-extension
7. Go to WooCommerce - Settings - Advanced - woocommerce.com and toggle Allow usage of WooCommerce to be tracked
8. See the event woocommerce_allow_tracking_toggled with context `settings`  and the right previous_value and new_value
9. Toggle it again 
10. See the event again working with the right values
11. Change a value that is not Allow usage of WooCommerce to be tracked
12. Nothing is triggered
13. toggle OFF Allow usage of WooCommerce to be tracked
14. Connect to WCCOM 
15. See the event again working with the right values and the context `wccom_connect`
16. Go to WooCommerce - Settings - Advanced - woocommerce.com and toggle OFF Allow usage of WooCommerce to be tracked
17. Force TrackingOptIn to appear in notes.
18. Click Activate usage tracking on this note
19.  See the event again working with the right values and the context `usage_tracking_note`
20. Verify that the option is not being toggled anywhere else.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
